gcs/src/main/scala/quasar/blobstore/gcs/SAAuthConfig.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.blobstore.gcs
+
+import scala.Predef.String
+
+import argonaut._, Argonaut._
+
+import cats.implicits._
+
+import scala.{Array, Byte, StringContext, Either}
+
+import java.net.{URI, URISyntaxException}
+
+final case class GoogleAuthConfig(authCfg: ServiceAccountConfig) {
+  import GoogleAuthConfig.serviceAccountConfigCodecJson
+  val serviceAccountAuthBytes: Array[Byte] = authCfg.asJson.toString.getBytes("UTF-8")
+}
+
+final case class ServiceAccountConfig(
+  tokenUri: URI,
+  authProviderCertUrl: URI,
+  privateKey: String,
+  clientId: String,
+  clientCertUrl: URI,
+  authUri: URI,
+  projectId: String,
+  privateKeyId: String,
+  clientEmail: String,
+  accountType: String)
+
+object GoogleAuthConfig {
+  val Redacted: String = "<REDACTED>"
+
+  implicit val uriCodecJson: CodecJson[URI] =
```

Comment:
FYI Normally json codecs are out of scope for async-blobstore implementations but it's useful in this case because of the tests making use of them (& argonaut is needed as a dep for gcs anyway).

Reply:
so when designing/implementing async-blobstore things we should avoid writing Codecs for objects? does that mean we should leave the Codecs to the actual connector? Just trying to get the big picture here.

Reply:
Yes, the other connectors don't have any dependency on argonaut or other json lib. The codecs do indeed reside in the datasource/destination.